File: count_stats.py
from pathlib import Path

def count_stats(file_path, config):
    if not check_file_permission(file_path, "read", config):
        raise PermissionError(f"Read access denied for {file_path}")
    file_path = Path(file_path).resolve()
    if not file_path.is_file():
        raise ValueError(f"File not found: {file_path}")
    with open(file_path, "r", encoding="utf-8") as f:
        content = f.read()
    lines = len(content.splitlines())
    words = len(content.split())
    chars = len(content)
    return f"Lines: {lines}, Words: {words}, Characters: {chars}"

File: ts-tools/count_stats.ts
import * as fs from 'fs';
import * as path from 'path';
import { checkFilePermission } from './check_file_permission';

export function countStats(filePath: string, config: any): Promise<string> {
  return new Promise((resolve, reject) => {
    checkFilePermission(filePath, 'read', config)
      .then(() => {
        const resolvedPath = path.resolve(filePath);
        fs.readFile(resolvedPath, 'utf-8', (err, data) => {
          if (err) {
            reject(err);
          } else {
            const lines = data.split('\n').length;
            const words = data.split(/\s+/).length;
            const chars = data.length;
            resolve(`Lines: ${lines}, Words: ${words}, Characters: ${chars}`);
          }
        });
      })
      .catch(reject);
  });
}
